uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/WorkPattern.php

```diff
@@ -438,4 +441,36 @@ public function links() {
       'id' => "{$workWeekTable}:pattern_id"
     ];
   }
+
+  /**
+   * Returns the type of this day for the contact whether its a Working Day, Non Working Day, Weekend or Public Holiday
+   * The method first of all finds the valid work pattern for this contact for the given date
+   * and also the effective start date of the contact work pattern
+   * This information is then used to find the work day type
+   *
+   * @param int $contactId
+   * @param \DateTime $date
+   *
+   * @return string
+   *   Can be any of the following:
+   *   - Public Holiday: When work day is a public holiday
+   *   - Weekend: When work day is weekend
+   *   - No: When work day is Non working day
+   *   - Yes: When work day is working day or All day
+   */
+  public static function getWorkDayType($contactId, DateTime $date) {
```

Comment:
So, after some thought I think this method should be on the ContactWorkPattern BAO. The reasons are:
- It already uses 2 of the ContactWorkPattern methods
- The WorkPattern itself doesn't know anything about the Contact, so it should not know how to fetch anything based on a contact ID
- Thinking in organization and communication terms, a contact doesn't know anything about a work pattern and, the same way, a work pattern doesn't know anything about a contact. The only one who knows both is the ContactWorkPattern, so this type of the glue should be on it